### PR TITLE
settings,*: give better names to certain settings

### LINKED ISSUE #109046: settings: some settings are poorly named and don't make a good story in docs
Extracted from #108508..

**Describe the problem**

Several settings have non-descriptive names. A couple even escaped the name linter when they were originally introduced due to a build system mishap.

We would like to rename settings for better UX and docs.

**Expected behavior**

New setting names, but without breaking backward compatibility.

Epic: CRDB-28893

Jira issue: CRDB-30749

### PR DESCRIPTION
**Pull request by knz**
Fixes #109046.

Now we have a framework that allows us to rename cluster settings
while preserving backward-compatibility, we can clean up the settings
whose name didn't pass the linter, and had been "grand-fathered". This
also simplifies the linter.

The following have been renamed:

| Previous name                                  | New name                                           |
|------------------------------------------------|----------------------------------------------------|
| `kv.raft_log.disable_synchronization_unsafe`   | `kv.raft_log.synchronization.disabled`             |
| `sql.trace.log_statement_execute`              | `sql.log.all_statements.enabled`                   |
| `cloudstorage.gs.chunking.retry_timeout`       | `cloudstorage.gs.chunking.per_chunk_retry.timeout` |
| `trace.debug.enable`                           | `trace.http_debug_endpoint.enabled`                |
| `bulkio.restore.memory_monitor_ssts`           | `bulkio.restore.sst_memory_limit.enabled`          |
| `bulkio.restore.use_simple_import_spans`       | `bulkio.restore.simple_import_spans.enabled`       |
| `bulkio.backup.export_request_verbose_tracing` | `bulkio.backup.verbose_tracing.enabled`            |
| `changefeed.idle_timeout`                      | `changefeed.auto_idle.timeout`                     |

The following name suffixes have been adjusted to follow our linter
guidelines:

```
bulkio.backup.split_keys_on_timestamps - added .enabled
changefeed.balance_range_distribution.enable - renamed to .enabled
changefeed.batch_reduction_retry_enabled - renamed to .enabled
changefeed.new_pubsub_sink_enabled - renamed to .enabled
changefeed.new_webhook_sink_enabled - renamed to .enabled
changefeed.permissions.require_external_connection_sink - added .enabled
debug.panic_on_failed_assertions - added .enabled
diagnostics.reporting.send_crash_reports - added .enabled
kv.closed_timestamp.follower_reads_enabled - renamed to .enabled
kv.range_merge.queue_enabled - renamed to .enabled
kv.range_split.by_load_enabled - renamed to .enabled
kv.transaction.parallel_commits_enabled - renamed to .enabled
kv.transaction.write_pipelining_enabled - renamed to .enabled
kv.transaction.write_pipelining_max_batch_size - renamed to .max_batch_size
server.clock.forward_jump_check_enabled - renamed to .enabled
server.oidc_authentication.autologin - added .enabled
server.web_session_timeout - renamed to .timeout
sql.distsql.flow_stream_timeout - renamed to .timeout
sql.metrics.statement_details.dump_to_logs - added .enabled
stream_replication.job_liveness_timeout - renamed to .timeout
```

Release note (cli change): The following user-visible cluster settings
have been renamed. The previous name is still available for backward
compatibility.

| Previous name                                  | New name                                             |
|------------------------------------------------|------------------------------------------------------|
| `server.web_session_timeout`                   | `server.web_session.timeout`                         |
| `kv.closed_timestamp.follower_reads_enabled`   | `kv.closed_timestamp.follower_reads.enabled`         |
| `kv.range_split.by_load_enabled`               | `kv.range_split.by_load.enabled`                     |
| `changefeed.balance_range_distribution.enable` | `changefeed.balance_range_distribution.enabled`      |
| `changefeed.batch_reduction_retry_enabled`     | `changefeed.batch_reduction_retry.enabled`           |
| `server.clock.forward_jump_check_enabled`      | `server.clock.forward_jump_check.enabled`            |
| `server.oidc_authentication.autologin`         | `server.oidc_authentication.autologin.enabled`       |
| `sql.metrics.statement_details.dump_to_logs`   | `sql.metrics.statement_details.dump_to_logs.enabled` |
| `sql.trace.log_statement_execute`              | `sql.log.all_statements.enabled`                     |
| `trace.debug.enable`                           | `trace.http_debug_endpoint.enabled`                  |

Epic: CRDB-28893

### CODE REVIEW

File: pkg/ccl/backupccl/backup_processor.go [hunk 1]
```diff
@@ -85,13 +85,15 @@ var (
 		"bulkio.backup.split_keys_on_timestamps",
 		"split backup data on timestamps when writing revision history",
 		true,
+		settings.WithName("bulkio.backup.split_keys_on_timestamps.enabled"),
```
Comment (stevendanna) [inline pkg/ccl/backupccl/backup_processor.go:88]: @dt Now that we have the option to rename things, I wonder if we want to make a follow-up pass to @knz's work here to address some of our long-standing setting naming gripes?

**Review by j82w — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109074)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @knz, @samiskin, and @yuzefovich)

___
*[`-- commits` line 4 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/109074#-NcN-jaRErM8YTSGib8O):*
I created this issue about possibly improving the naming. What do you think? https://github.com/cockroachdb/cockroach/issues/108508
___
*[`-- commits` line 10 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/109074#-NcN-R1P98AAv0P_tUzW):*
Shouldn't there be a release note that cluster settings are getting aliased with new names?

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109074)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @j82w, @samiskin, and @yuzefovich)

___
*[`-- commits` line 4 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/109074#-NcN-jaRErM8YTSGib8O):*
<details><summary><i>Previously, j82w (Jake) wrote…</i></summary><blockquote>

I created this issue about possibly improving the naming. What do you think? https://github.com/cockroachdb/cockroach/issues/108508
</blockquote></details>

Yes I saw. I started this work in light of your issue.
However, note that the concern you raised there already has a plan under epic CRDB-6671. There's multiple steps there before we can consider directly what you suggest.
___
*[`-- commits` line 10 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/109074#-NcN-R1P98AAv0P_tUzW):*
<details><summary><i>Previously, j82w (Jake) wrote…</i></summary><blockquote>

Shouldn't there be a release note that cluster settings are getting aliased with new names?
</blockquote></details>

I thought I included this release note already. Doesn't it show up for you?

**Comment by j82w** [comment]
*[`-- commits` line 10 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/109074#-NcN-R1P98AAv0P_tUzW):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

I thought I included this release note already. Doesn't it show up for you?
</blockquote></details>

The 2nd commit has the release note. I left the comment looking at the first commit and forgot to delete it.

**Comment by j82w** [comment]
*[`-- commits` line 4 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/109074#-NcN-jaRErM8YTSGib8O):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

Yes I saw. I started this work in light of your issue.
However, note that the concern you raised there already has a plan under epic CRDB-6671. There's multiple steps there before we can consider directly what you suggest.
</blockquote></details>

If it already supports alias and a rename is being done why can't we incorporate the setting type into the name now?

**Comment by knz** [comment]
> If it already supports alias and a rename is being done why can't we incorporate the setting type into the name now?

Because the recommendation you made is only one of the ways to solve the problem that you identified. We are going to explore other ways to solve it first.

**Review by ericharmeling — APPROVED** [review]
Great usability improvement. We'll definitely want docs follow-up on this.

@florence-crl

**Pull request merged**